File: dart/beer-song/lib/beer_song.dart
class BeerSong {
  List<String> recite(num start, num numVerses) {
    List<String> res = [];
    for (num i = 0; i < numVerses; i++) {
      num currBottles = start - i;
      String numBottles = currBottles == 0 ? "No more" : "${currBottles}";
      String numBottlesLower = currBottles == 0 ? "no more" : "${currBottles}";
      String oneBottle = currBottles == 1 ? "it" : "one";
      String bottlesRemaining = currBottles == 1 ? 'no more' : "${currBottles - 1}";
      res.add("${numBottles} bottle${currBottles == 1 ? '' : 's'} of beer on the wall, ${numBottlesLower} bottle${currBottles == 1 ? '' : 's'} of beer.");
      res.add(currBottles == 0 ? (
        "Go to the store and buy some more, 99 bottles of beer on the wall."
      ) : "Take ${oneBottle} down and pass it around, ${bottlesRemaining} bottle${currBottles == 2 ? '' : 's'} of beer on the wall.");
      if (numVerses > 0 && i != numVerses - 1) {
        res.add('');
      }
    }
    return res;
  }
}

File: typescript/beer-song/beer-song.ts
export default class BeerSong {
  static verse(n: number): string {
    const numBottles = n === 0 ? 'No more' : `${n}`
    let res = `${numBottles} bottle${n === 1 ? '' : 's'} of beer on the wall, ${numBottles.toLowerCase()} bottle${n === 1 ? '' : 's'} of beer.\n`
    res += n === 0 ? (
      'Go to the store and buy some more, 99 bottles of beer on the wall.\n'
    ) : `Take ${n === 1 ? 'it' : 'one'} down and pass it around, ${n == 1 ? 'no more' : `${n - 1}`} bottle${n === 2 ? '' : 's'} of beer on the wall.\n`
    return res;
  }

  static sing(start: number = 99, end: number = 0): string {
    let res: string[] = []
    for (let i = start; i >= end; i--) {
      res.push(BeerSong.verse(i))
    }
    return res.join('\n')
  }
}